File: src/main/java/tn/esprit/projet/models/Planification.java
package tn.esprit.projet.models;

import java.util.Objects;

public class Planification {

    private int idPlanification;
    private int idEvent;
    private String description;
    private String duree;

    // Relation objet avec Event
    private Event event;

    public Planification() {}

    public Planification(int idPlanification, int idEvent, String description, String duree) {
        this.idPlanification = idPlanification;
        this.idEvent = idEvent;
        this.description = description;
        this.duree = duree;
    }

    public int getIdPlanification() { return idPlanification; }
    public void setIdPlanification(int idPlanification) { this.idPlanification = idPlanification; }

    public int getIdEvent() { return idEvent; }
    public void setIdEvent(int idEvent) { this.idEvent = idEvent; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getDuree() { return duree; }
    public void setDuree(String duree) { this.duree = duree; }

    public Event getEvent() { return event; }
    public void setEvent(Event event) { this.event = event; }

    @Override
    public String toString() {
        return "Planification{" +
                "idPlanification=" + idPlanification +
                ", idEvent=" + idEvent +
                ", description='" + description + '\'' +
                ", duree='" + duree + '\'' +
                ", event=" + event +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Planification that)) return false;
        return idPlanification == that.idPlanification && idEvent == that.idEvent &&
                Objects.equals(description, that.description) && Objects.equals(duree, that.duree);
    }

    @Override
    public int hashCode() {
        return Objects.hash(idPlanification, idEvent, description, duree);
    }
}

File: src/Entity/Planification.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

use App\Repository\PlanificationRepository;

#[ORM\Entity(repositoryClass: PlanificationRepository::class)]
#[ORM\Table(name: 'planification')]
class Planification
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id_planification = null;

    public function getId_planification(): ?int
    {
        return $this->id_planification;
    }

    public function setId_planification(int $id_planification): self
    {
        $this->id_planification = $id_planification;
        return $this;
    }

    #[ORM\ManyToOne(targetEntity: Event::class, inversedBy: 'planifications')]
    #[ORM\JoinColumn(name: 'id_event', referencedColumnName: 'id')]
    private ?Event $event = null;

    public function getEvent(): ?Event
    {
        return $this->event;
    }

    public function setEvent(?Event $event): self
    {
        $this->event = $event;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $description = null;

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: true)]
    private ?int $duree = null;

    public function getDuree(): ?int
    {
        return $this->duree;
    }

    public function setDuree(?int $duree): self
    {
        $this->duree = $duree;
        return $this;
    }

}
